### PR TITLE
update to MGXS in tools

### PR DESCRIPTION
**Pull request by ragusa**
- Adding 2 more examples of MGXS generation using OpenMC
- Restructuring the `tools` folder to have `users` and `developers` subfolder
- Adding details to readme files; adding subfolder readme files.

### CODE REVIEW

File: tools/README.md [hunk 1]
```diff
@@ -1,11 +1,33 @@
-### Contents of `tools/`
+# Contents 
 
-This directory contains various tools for use by both developers and users.
-A description of each of the files in this directory is given in the table below.
+This `tools/` directory contains various tools for use by both developers and users.
```
Comment (zhardy-lanl) [inline tools/README.md:3]:
```suggestion
This `tools` directory contains various tools for use by both developers and users.
```

File: tools/README.md [hunk 2]
```diff
@@ -1,11 +1,33 @@
...
+
+Files located at the root of `tools/`:
```
Comment (zhardy-lanl) [inline tools/README.md:5]:
```suggestion
Files located at the root of `tools`:
```

File: tools/users/xs/MGXS_README.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# Multigroup Cross-section Generation Using OpenMC
```
Comment (zhardy-lanl) [inline tools/users/xs/MGXS_README.md:1]:
```suggestion
# Multigroup Cross-Section Generation Using OpenMC
```

File: tools/developers/ignorelist.txt
Comment (zhardy-lanl) [inline tools/developers/ignorelist.txt:1]: What does this file do?

**Review by zhardy-lanl — CHANGES_REQUESTED** [review]

File: tools/developers/ignorelist.txt
Comment (ragusa) [reply tools/developers/ignorelist.txt:1]: I do not know. I think it's related to boost, maybe. @wdhawkins may have committed it. I just renamed the sub-folder `developer` to `developers`. I may need a description in a readme file. If so, @wdhawkins can fix this after this is merged. Maybe an issue should be created so as to not forget.

Comment (ragusa) [reply tools/developers/ignorelist.txt:1]: I will resolve this now, as this may not be part of this PR.

**Review by andrsd — APPROVED** [review]

**Comment by wdhawkins** [comment]
With the move of `lsan.supp` to and `ignorelist.txt` to `tools/developers`, you'll also need to update the nightly sanitizer workflow.

**Comment by ragusa** [comment]
> With the move of `lsan.supp` to and `ignorelist.txt` to `tools/developers`, you'll also need to update the nightly sanitizer workflow.

Ok. I start to understand the use of those files. Can you write pointers on where I need to look? Thanks.

**Comment by wdhawkins** [comment]
> > With the move of `lsan.supp` to and `ignorelist.txt` to `tools/developers`, you'll also need to update the nightly sanitizer workflow.
> 
> Ok. I start to understand the use of those files. Can you write pointers on where I need to look? Thanks.

`lsan.supp` is used in the `sanitize` target in `.github/workflows/builds.yaml`. For both the `test` and `test tutorials` steps you'll want to change the `export LSAN_OPTIONS=suppressions=$PWD/tools/developer/lsan.supp` line. `ignorelist.txt` is used in `CMakePresets.json` in the sanitizer compile target. You'll need to change the `CMAKE_CXX_FLAGS` and `CMAKE_C_FLAGS` lines.

**Comment by ragusa** [comment]
@wdhawkins : 
1. should Iremove `xs` out of `users` and therefore leaving `scaling` where it is, so that the subfolder structure is `developers`, `xs`, `scaling` ?
2. or should I move `scaling` inside either `users'` or `developers`?

I prefer #1. If we go with #1, do you want me to revert `developers`back to `developer`?

**Comment by wdhawkins** [comment]
> @wdhawkins :
> 
>     1. should I remove `xs` out of `users` and therefore leave `scaling` where it is, so that the subfolder structure is `developers`, `xs`, `scaling` ?
> 
>     2. or should I move `scaling` inside either `users` or `developers`?
> 
> 
> I prefer option 1. If we go with option 1, do you want me to revert `developers`back to `developer`?

Option 1 sounds good to me as well. I'm also ok reverting `developers` to `developer`.

File: tools/developers/ignorelist.txt
Comment (ragusa) [reply tools/developers/ignorelist.txt:1]: I've added a blurb explaining the `ignorelist.txt` file

**Comment by ragusa** [comment]
@zhardy-lanl Updates are done. Ready for your re-review.

File: tools/README.md [hunk 4]
```diff
@@ -1,12 +1,42 @@
...
+### Tools for Developer
+
+Files are located in `tools/developer`.
+
+A description of the files or sub-directory is given in the table below.
+
+| File or Directory     | Description                                             |
+|-----------------------|---------------------------------------------------------|
+| `lsan.supp`           | Leak sanitizer suppression file (used in git workflows) |
+| `ignorelist.txt` | (used in CMakepresets.json)                             |
```
[the remark this answers is not in the document]
Comment (ragusa) [reply tools/README.md:18]: i am sorry. i didn't fully understand the comment. I think you want me to align the `|`. it's done.

[the remark this answers is not in the document]
Comment (zhardy-lanl) [reply tools/README.md]: My comment was unrelated to the suggested code change, haha!

**Review by zhardy-lanl — APPROVED** [review]

File: tools/README.md [hunk 5]
```diff
@@ -1,12 +1,42 @@
...
+### Tools for Developer
+
+Files are located in `tools/developer`.
+
+A description of the files or sub-directory is given in the table below.
+
+| File or Directory  | Description                                             |
+|--------------------|---------------------------------------------------------|
+| `lsan.supp`        | Leak sanitizer suppression file (used in git workflows) |
+| `ignorelist.txt`   | (used in CMakepresets.json)                             |
```
Comment (wdhawkins) [inline tools/README.md:18]: List of source-level entities that should be ignored by sanitizers.

**Review by wdhawkins — APPROVED** [review]

File: tools/README.md (hunk 5 above)
Comment (ragusa) [reply tools/README.md:18]: added text

**Comment by ragusa** [comment]
partially closes #476

**Pull request merged**